plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -127,6 +128,10 @@ def set_replica_target
       template[:spec][:replicas] = @doc.replica_target if template[:kind] == 'Deployment'
     end
 
+    def set_name
+      template[:metadata][:name] = @doc.kubernetes_role.resource_name
```

Comment:
What is metadata.name? deployment.metadata.name? (or job.metadata.name or daemonset...)?

I mean, in terms of ```kubectl explain```, what do you refer to? Just to be sure I understand